soco/core.py

```diff
@@ -1833,6 +1873,185 @@ def add_item_to_sonos_playlist(self, queueable_item, sonos_playlist):
+
+            # object_id specified by the item_id
+            obj.reorder_sonos_playlist('SQ:0', tracks=[0, ], new_pos=[1, ])
+            # OR object_id specified by the DidlPlaylistContainer object
+            spl = obj.get_sonos_playlists()[0]
+            obj.reorder_sonos_playlist(spl, tracks=[0, ], new_pos=[1, ])
+
+          To delete the second track::
+
+            # tracks/new_pos are a list of int
+            obj.reorder_sonos_playlist('SQ:0', tracks=[1, ], new_pos=[None, ])
+            # OR tracks/new_pos are a list of int-like
+            obj.reorder_sonos_playlist('SQ:0', tracks=['1', ], new_pos=['', ])
+            # OR tracks/new_pos are strings - no transform is done
+            obj.reorder_sonos_playlist('SQ:0', tracks='1', new_pos='')
+
+        Args:
+            object_id (str): The item_id of the Sonos playlist or the Sonos
+                playlist object
+                (:py:class:`~.soco.data_structures.DidlPlaylistContainer`).
+            tracks: (list): list of track indices(int) to reorder. May also be
+                a list of int like things. i.e. ``['0', '1',]`` OR it may be a
+                str of comma separated int like things. ``"0,1"``.  Tracks are
+                **0**-based. Meaning the first track is track 0, just like
+                indexing into a Python list.
+            new_pos (list): list of new positions (int|None)
+                corresponding to track_list. MUST be the same type as
+                ``tracks``. **0**-based, see tracks above. ``None`` is the
+                indicator to remove the track. If using a list of strings,
+                then a remove is indicated by an empty string.
+            update_id (int): operation id (default: 0) If set to 0, a lookup
+                is done to find the correct value.
+
+        Returns:
+            tuple: A  tuple(int, int, int) containing 3 values:
+                Change in size between original playlist and the resulting
+                playlist, the length of resulting playlist, and the new
+                update_id.
+
+        Raises:
+            SoCoUPnPException: If object_id does not exist.
+        """
+        # allow either a string 'SQ:10' or an object with item_id attribute.
+        object_id = getattr(object_id, 'item_id', object_id)
+
+        if isinstance(tracks, UnicodeType):
+            track_list = [tracks, ]
+            position_list = [new_pos, ]
+        elif type(tracks) is int:
+            track_list = [tracks, ]
+            position_list = [new_pos or '', ]
+        else:
+            track_list = [str(x) for x in tracks]
+            position_list = [str(x) if x is not None else '' for x in new_pos]
+
+        if update_id == 0:  # retrieve the update id for the object
+            response, _ = self._music_lib_search(object_id, 0, 1)
+            update_id = response['UpdateID']
+        change = 0
+
+        for track, position in zip(track_list, position_list):
```

Comment:
Here we need to careful. When I wrote above, that I was unsure if you handled the non-iterable case correctly, it was because I was unsure if the under lying call takes a single position or a list of position. I now see that it takes a list, which of course means that the non-iterable input case is correct.

However, if it does take a list, I would highly suggest to turn all iterable input args into a list and execute only one request. The reason for this is two-fold:

 1. Performance, we should of course never prematurely optimize, but I think this just makes sense. For apps doing network IO, that will always be the "slow thing" and since it is made so easy to do all at once we might as well do it.
 2. Deletions. The significantly more important issue, is that if we do only do one request, we can hope that everything will also be executed as one query on the other end and this is important for deletions. We went through this somewhere else, where the call only took one argument, maybe it was for queue manipulation. Consider this, if I ask this method to delete tracks 0 and 3. After having deleted track 0, the second track that I wanted to delete is now in position 2 **not** 3. This can be fixed by sorting deletions descending, but here, at least of it works, it would be soooo much easier to just do one request. It may also be, that they actually iterate on the other end, in which case, we have to handle this problem here.

Deletions stuff, and the example above, should be tested.

The one thing that I do not know is, if the list form accepts mixing operations, move and delete, or if it might separate them out on the other end. This will make everything more complicated.